docs/releases/3.0.txt

```diff
@@ -312,6 +312,13 @@ Models
 
 * :attr:`.FileField.upload_to` now supports :class:`pathlib.Path`.
 
+* The new ``allows_group_by_selected_pks_on_model()`` method of
+  ``django.db.backends.base.BaseDatabaseFeatures`` allows using an optimization
+  to ``GROUP BY`` only by selected models' primary keys for unmanaged models on
+  PostgreSQL, which is disabled by default. If you want to use this feature you
+  have to subclass the PostgreSQL database engine and override
+  ``allows_group_by_selected_pks_on_model()``.
```

Comment:
I think this sentence isn't enough. 

* You have to override the method on the features class, 
* Then, override the database wrapper, 
* Then, save it in the right plave, i.e. a `my_backend.base`, pointing `DATABASES` (`BACKEND`) to `my_backend`. 

I think we need an example of that. 

SO here, broken up for consideration: 

```
The new ``allows_group_by_selected_pks_on_model()`` method of
``django.db.backends.base.BaseDatabaseFeatures`` allows optimization of ``GROUP
BY`` clauses to require only the selected models' primary keys.

This optimization is currently only supported on PostgreSQL. By default,
it is only applied to **managed models**.

To enable the the GROUP BY primary key-only optimization for unmanaged models,
subclass the PostgreSQL database engine, overriding the features class
``allows_group_by_selected_pks_on_model()`` as you require. See LINK TO FULL
EXAMPLE for an example.
```

Then in FULL EXAMPLE we need the full example @Tasssadar came up with [in this comment](https://github.com/django/django/pull/11606#issuecomment-517255305), plus the file to save it in, and the `DATABASES` setting example. 

Open to suggestions as to where but I would create a new section in `databases.txt` called, `Subclassing the built-in backends`, just before `Using a 3rd-party database backend`, to hold FULL EXAMPLE (and be the target of LINK TO FULL EXAMPLE).

Reply:
@Tasssadar: Do you this you have bandwidth to make these docs changes? (Thanks if you do! 🎉)

Reply:
Back from vacation, I tried to write an example for the docs, please take a look :)